clang/lib/DPCT/SaveNewFiles.cpp

```diff
@@ -220,6 +236,40 @@ static bool checkOverwriteAndWarn(StringRef OutFilePath, StringRef InFilePath) {
 void processallOptionAction(clang::tooling::UnifiedPath &InRoot,
                             clang::tooling::UnifiedPath &OutRoot,
                             bool IsForSYCL) {
+  extern DpctOption<clang::dpct::opt, bool> ProcessAll;
+  if (ProcessAll) {
+    std::error_code EC;
+    for (fs::recursive_directory_iterator Iter(Twine(InRoot.getPath()), EC),
+         End;
+         Iter != End; Iter.increment(EC)) {
+      if ((bool)EC) {
+        std::string ErrMsg = "[ERROR] Access : " + std::string(InRoot.getPath()) +
+                             " fail: " + EC.message() + "\n";
+        PrintMsg(ErrMsg);
+        continue;
+      }
+      if (Iter->type() == fs::file_type::symlink_file) {
+        tooling::UnifiedPath FilePath = Iter->path();
+        std::string TargetPath = FilePath.getCanonicalPath().str();
+        std::string LinkPath = FilePath.getAbsolutePath().str();
+        if (IncludeFileMap[FilePath]) {
+          rewriteFileName(TargetPath, TargetPath);
+          rewriteFileName(LinkPath, LinkPath);
+        }
+        rewriteDir(TargetPath, InRoot, OutRoot);
+        rewriteDir(LinkPath, InRoot, OutRoot);
+        if (!sys::fs::exists(LinkPath)) {
+          std::error_code ec = sys::fs::create_link(TargetPath, LinkPath);
+          if (ec) {
+            std::cerr << "Error creating symlink: " << ec.message()
```

Comment:
should not use std::cerr directly for this error message,  orelse --output-log option will not work as expected.